dashboard/app/models/unit_group.rb

```diff
@@ -259,32 +273,22 @@ def assignable_info(user = nil)
   end
 
   def self.all_courses
-    all_courses = Rails.cache.fetch('valid_courses/all') do
-      UnitGroup.all.to_a
-    end
-    all_courses.freeze
+    return all.to_a unless should_cache?
+    @@all_courses ||= course_cache.values.uniq.freeze
   end
 
   def self.family_names
     CourseVersion.course_offering_keys('UnitGroup')
   end
 
-  # Get the set of valid courses for the dropdown in our sections table. This
-  # should be static data for users without any course experiments enabled, but
-  # contains localized strings so we can only cache on a per locale basis.
-  #
+  # Get the set of valid courses for the dropdown in our sections table.
   # @param [User] user Whose experiments to check for possible unit substitutions.
+  # @return [Array<UnitGroup>]
   def self.valid_courses(user: nil)
-    # Do not cache if the user might have a course experiment enabled which puts them
-    # on an alternate unit.
```

Comment:
note to other reviewers -- this comment became obsolete when we stopped caching translated content in the rails cache at this point in history: https://github.com/code-dot-org/code-dot-org/commit/ed1317a2d42c6a5f451f4fd3a77768c4066865fc#diff-18b9a246b968bfc5cdf7e690478c4b62144716bd2d5fc57463a431c9bb015129R242